XXX-asyncwrap-api.md

```diff
@@ -0,0 +1,309 @@
+| Title  | AsyncWrap API |
+|--------|---------------|
+| Author | @trevnorris   |
+| Status | DRAFT         |
+| Date   | 2016-04-05    |
+
+## Description
+
+Since its initial introduction with the `AsyncListener` API, `AsyncWrap` has
+slowly evolved to ensure a generalized API that would serve as a solid base for
+module authors who wished to add events to the life cycle of the event loop.
```

Comment:
I don't think telling someone to just go `dig through node core while we're working out a potential API for it` is really all that helpful (or friendly). @mike-kaufman appears to be making the point that while this new API appears useful it's not clear given the description in this eps exactly *how* it would be used and what benefit it brings. Several examples and a description of the lifecycle of a hook would make that *much* clearer. No one is asking for 100% of things to be documented for end users. What is being asked for is a bit more clarity... and I think a bit more patience would likely be a good thing also.